src/MongoDB.Driver/MongoClient.cs

```diff
@@ -328,38 +328,26 @@ public override IMongoClient WithWriteConcern(WriteConcern writeConcern)
         // private methods
         private bool AreSessionsSupported(CancellationToken cancellationToken)
         {
-            return AreSessionsSupported(_cluster.Description) ?? AreSessionsSupportedAfterServerSelection(cancellationToken);
+            return _cluster.Description.AreSessionsSupported ?? AreSessionsSupportedAfterServerSelection(cancellationToken);
         }
 
         private async Task<bool> AreSessionsSupportedAsync(CancellationToken cancellationToken)
         {
-            return AreSessionsSupported(_cluster.Description) ?? await AreSessionsSupportedAfterSeverSelctionAsync(cancellationToken).ConfigureAwait(false);
-        }
-
-        private bool? AreSessionsSupported(ClusterDescription clusterDescription)
```

Comment:
The logic for determining whether sessions are supported or not based on the information in a ClusterDescription was centralized in the ClusterDescription class itself.